frontend/src/metabase/visualizations/visualizations/PieChart/utils.ts

```diff
@@ -0,0 +1,23 @@
+export function getMaxLabelDimension(
+  d3Arc: d3.svg.Arc<d3.svg.arc.Arc>,
+  slice: d3.svg.arc.Arc,
+  index = 0,
+) {
+  const innerRadius = d3Arc.innerRadius()(slice, index);
+  const outerRadius = d3Arc.outerRadius()(slice, index);
+  const arcWidth = outerRadius - innerRadius;
+
+  const arcAngle =
+    d3Arc.startAngle()(slice, index) - d3Arc.endAngle()(slice, index);
+
+  // using law of cosines to calculate the arc length
+  // c = sqrt(a^2 + b^2﹣2*a*b * cos(arcAngle))
+  // where a = b = innerRadius
+
+  const arcLength = Math.sqrt(
```

Comment:
nit: The name gave me the impression that `arcLength` measure the length of the arc (along the curve = `2*Pi*innerRadius * (arcAngle/(2*Pi))`), but this seems to measure the displacement between the 2 end coordinates that represent the inner arc. I don't know what is the math term for this, but I'm looking for something along the line "the displacement between the 2 points that are the ending of this arc".